### PR TITLE
Migrated `databasebackups` datasource to Framework

### PR DESCRIPTION
**Pull request by ezilber-akamai**
## 📝 Description

Migrated `databasebackups` datasource to Framework

## ✔️ How to Test

`make PKG_NAME="linode/databasebackups" testacc`

TPT-1967

### CODE REVIEW

**Review by yec-akamai — APPROVED** [review]

File: linode/databasebackups/datasource_test.go [hunk 1]
```diff
@@ -23,65 +20,28 @@ var (
...
-		Created: &currentTime,
-	}
-
-	result := databasebackups.FlattenBackup(backup)
-	if result["id"] != backup.ID {
-		t.Fatal(cmp.Diff(result["id"], backup.ID))
-	}
-
-	if result["label"] != backup.Label {
-		t.Fatal(cmp.Diff(result["label"], backup.Label))
```
Comment (zliang-akamai) [inline linode/databasebackups/datasource_test.go:50]: Was this test case deleted for some reasons?

Comment (zliang-akamai) [reply linode/databasebackups/datasource_test.go:50]: I think we can make `parseMySQLBackup` and `parsePostgresSQLBackup` as independent functions and call them from `parseXXXXBackups` in the model file, and then bring these tests back to test those functions.

File: linode/databasebackups/framework_datasource.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+			ctx, r.client, data.Filters, listMySQLBackups, data.Order, data.OrderBy)
+		if d != nil {
+			resp.Diagnostics.Append(d)
+			return
+		}
+
+		data.parseMySQLBackups(helper.AnySliceToTyped[linodego.MySQLDatabaseBackup](result))
+	}
+
+	if data.DatabaseType.ValueString() == "postgres" {
```
Comment (zliang-akamai) [inline linode/databasebackups/framework_datasource.go:100]:
```suggestion
	} else if data.DatabaseType.ValueString() == "postgres" {
```

File: linode/databasebackups/framework_datasource.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	listPostgresSQLBackups := func(ctx context.Context, client *linodego.Client, filter string) ([]any, error) {
+		backups, err := client.ListPostgresDatabaseBackups(ctx, int(data.DatabaseID.ValueInt64()), &linodego.ListOptions{
+			Filter: filter,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		return helper.TypedSliceToAny(backups), nil
+	}
```
Comment (zliang-akamai) [inline linode/databasebackups/framework_datasource.go:87]: This is an optional suggestion.
I feel moving the function definition into the if else block and only define listMySQLBackups when the db type is mysql might improve the memory usage slightly, what do you think?

**Comment by zliang-akamai** [comment]
Overall looks good, nice work!

File: linode/databasebackups/framework_datasource.go (hunk 3 above)
Comment (ezilber-akamai) [reply linode/databasebackups/framework_datasource.go]: Good idea. I implemented this change!

File: linode/databasebackups/datasource_test.go (hunk 1 above)
Comment (ezilber-akamai) [reply linode/databasebackups/datasource_test.go:50]: As far as adding additional test cases, none of the parsing functions return anything so I'm not sure about the best way to test these.

Comment (zliang-akamai) [reply linode/databasebackups/datasource_test.go:50]: I think you can check the attributes in the model instance against the attributes in the linodego object

Comment (zliang-akamai) [reply linode/databasebackups/datasource_test.go:50]: I think this is optional because we didn't test model parsing in other data sources...

https://github.com/linode/terraform-provider-linode/actions/runs/5258218942
Let's wait for the action result, and you can decide whether to bring these tests back or not.

Comment (ezilber-akamai) [reply linode/databasebackups/datasource_test.go:50]: The action result will fail due to an unresolved issue with polling. Still trying to figure it out.

Comment (ezilber-akamai) [reply linode/databasebackups/datasource_test.go]: Solved!
